docs/reference/how-to/size-your-shards.asciidoc

```diff
@@ -68,7 +68,20 @@ This decreases the number of segments, which means less metadata is kept in
 heap memory.
 
 Every mapped field also carries some overhead in terms of memory usage and disk
-space. By default {es} will automatically create a mapping for every field in
+space. While the exact amount resource usage of each additional field depends
+on the type of field, an additional 1KB in memory use for each additional
+mapped field on data nodes is a good approximation in most cases.
+As a result, the more mapped fields you have per index, the fewer
+indices and shards each individual data node will be able to hold,
+all other things being equal. In addition to the per field overhead you should
+have heap available for the base Elasticsearch heap usage as well as the heap
+usage of the workload. For a small workload, 512MB of heap in addition to the
+per field overhead suffice in many cases. For tight workloads, it is possible to
+go even lower.
+For example, holding 1000 indices that each contain 4000 fields will require
```

Comment:
I sort of wonder whether the concrete figures given here should be another `Aim for` section below? Maybe later, just recording this thought for posterity here.

> Aim for 1000000 fields or fewer per GB of heap memory on data nodes, and account for overhead